notes/compilation-and-linkage.md

```diff
@@ -0,0 +1,729 @@
+
+After the generation of each translation unit, the translation units are then processed a few more times, but the most notable intermediate processing that occurs is the translation of the files from C/C++ to Assembly code, or .asm. These are very basic instructions that talk directly to the registers of your computer how to run your code (You'll get to see it if you look it up or if you have/will take CSE 351).
+
+After this, the each translation unit is then converted to binary, which is actually understandable by the machine itself. These are stored into Object files (.o or .obj), which is what any compiler will need to run code.
+
+To both preprocess and compile, there are defined commands to help you do this. For C:
+
+<div style="text-align: center">
+
+`gcc [optionalFlags] -c yourCFile.c -o outputObjectFile.o`
+
+</div>
+
+For C++:
+
+<div style="text-align: center">
+
+`g++ [optionalFlags] -c yourCFile.cpp -o outputObjectFile.o`
+
+</div>
+
+Both will run the preprocessor (or some derivative of that) and compiler your objects at the same time.
+
+## Static Linking
+
+Back to the bridge analogy, your manufacturers have now created each part! However, the functionality of some of the parts does not make sense individually. There might be a protrusion, hole, fillet, or chamfer that each manufacture made, but they don't really know what the purpose of it. You do though, those features are supposed to function with the other parts, so your job now is to tell them how to assemble the bridge, how to use each feature they made. 
+
+The final step in the compilation process, called linking, will assemble all your object files together into a single executable (.exe), and then you'll be perfect to run. We must bind all files together and map all necessary definitions to memory before the computer can run the program. When that's done, all the definitions and instructions in your code will be mapped to a memory location in your computer, and all translation units that interact with any given definition/instruction will have knowledge over them via the memory address (forshadowing).
+
+To do this, your compiler will need to assemble all your created object files, along with any object files related to the libraries you used, to create this executable. In Linux, we can once again do this via the command prompt. For C:
+
+<div style="text-align: center">
+
+`gcc [optionalFlags] -o executableName objectFile1.o objectFile2.o ... objectFileN.o [otherLibraries]`
+
+</div>
+
+For C++:
+
+<div style="text-align: center">
+
+`g++ [optionalFlags] -o executableName objectFile1.o objectFile2.o ... objectFileN.o [otherLibraries]`
```

Comment:
Could you give an example of a real command? It might help make things clear to people that haven't done this before